File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-4.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        counts = {}
        i = 0
        maxf = 0
        maxLen = 0
        for j in range(len(s)):
            if s[j] not in counts:
                counts[s[j]] = 1
            else:
                counts[s[j]] += 1
            maxf = max(maxf, counts[s[j]])

            while((j - i + 1) - maxf > k):
                counts[s[i]] -= 1
                i += 1
            
            maxLen = max(maxLen, j - i + 1)
        return maxLen

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-4.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> uniqueNums = unordered_set(nums.begin(), nums.end());
        int maxLen = 0;
        for(const int& n : uniqueNums){
            if(uniqueNums.contains(n-1)) continue;
            
            int length = 1;
            while(uniqueNums.contains(n+length)){
                length++;
            }
            maxLen = max(maxLen, length);
        }
        return maxLen;
    }
};
